sklearn/linear_model/tests/test_ridge.py

```diff
@@ -836,6 +836,65 @@ def test_ridge_fit_intercept_sparse():
         assert_array_almost_equal(dense.coef_, sparse.coef_)
 
 
+@pytest.mark.parametrize('return_intercept', [False, True])
+@pytest.mark.parametrize('sample_weight', [None, np.ones(1000)])
+@pytest.mark.parametrize('arr_type', [np.array, sp.csr_matrix])
+@pytest.mark.parametrize('solver', ['auto', 'sparse_cg', 'cholesky', 'lsqr',
+                                    'sag', 'saga'])
+def test_ridge_regression_check_arguments_validity(return_intercept,
+                                                   sample_weight, arr_type,
+                                                   solver):
+    """check if all combinations of arguments give valid estimations"""
+
+    # test excludes 'svd' solver because it raises exception for sparse inputs
+
+    X = np.random.rand(1000, 3)
+    true_coefs = [1, 2, 0.1]
+    y = np.dot(X, true_coefs)
+    X_testing = arr_type(X)
+
+    target = ridge_regression(X_testing, y, 1,
+                              solver=solver,
+                              sample_weight=sample_weight,
+                              return_intercept=return_intercept
+                              )
+    try:
```

Comment:
why do you need a try here? can we avoid it?

Reply:
replaced with if..else...